### PR TITLE
#29704 :- Submit search button in form-mini should be disabled until minimum search length is reached

### LINKED ISSUE #29704: Submit search button in form-mini should be disabled until minimum search length is reached
### Preconditions (*)

1. Base install of Magento 2.4.0 with sample data installed.
2. The issue is prevalent on every operating system and browser.

### Steps to reproduce (*)
1. Focus on form-mini search box.
2. Type one or two alphanumeric characters in search box.

### Expected result (*)
- Submit button (magnifying glass) remains disabled until minimum search length is reached:
Screen Shot 2020-08-20 at 2 34 09 PM-disabled

### Actual result (*)
- Submit button (magnifying glass) becomes enabled even if minimum search length is not reached:
Screen Shot 2020-08-20 at 2 34 09 PM

### Additional Info
Minimum search length is defined in the related component at https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Search/view/frontend/web/js/form-mini.js#L33

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ X ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR will fix the issue with the submit search button is not respecting the minimum search length
### Fixed Issues (if relevant)

1. Fixes magento/magento2#29704

### Manual testing scenarios (*)

1. Focus on form-mini search box.
2. Type one or two alphanumeric characters in search box.

### Expected Result 
image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by konarshankar07** [comment]
@magento run all tests

**Comment by rogyar** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rogyar. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rogyar, here is your [new Magento instance](https://pr-29724.instances.magento-community.engineering).
Admin access: [https://pr-29724.instances.magento-community.engineering/admin_34a9](https://pr-29724.instances.magento-community.engineering/admin_34a9)
Login: `ab5342ec` Password: `6759de2048d1`
Instance will be terminated in up to 3 hours.

**Comment by rogyar** [comment]
Hi @konarshankar07. Thank you for the fix. 

I'm just curious, how hard do you think will be to spread the same fix for the submit by "return" button? With your fix, the button stays disabled but it's still possible to submit the search query by pressing the "return" button. If this part of the fix will be complicated, feel free to create a separate PR. If the mentioned fix is easy to implement, it would be great to have it as a part of this PR. 

In any case, I would ask you to cover this fix by a simple MFTF test.  

Thank you!

**Comment by konarshankar07** [comment]
@magento run all tests

File: app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchButtonDisabledTillMinimumSearchLengthHit.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (gabrieldagama) [inline app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchButtonDisabledTillMinimumSearchLengthHit.xml]: Can we add the `Test` suffix here to follow the convention?

File: app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchButtonNotDisabledAfterMinimumSearchLengthHit.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontVerifySearchButtonNotDisabledAfterMinimumSearchLengthHit">
```
Comment (gabrieldagama) [inline app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchButtonNotDisabledAfterMinimumSearchLengthHit.xml]: Can we add the `Test` suffix here to follow the convention?

File: app/code/Magento/Search/Test/Mftf/ActionGroup/AssertStorefrontVerifySearchButtonIsNotDisabledActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <!-- Filter by search query and select -->
+    <actionGroup name="AssertStorefrontVerifySearchButtonIsNotDisabledActionGroup">
```
Comment (gabrieldagama) [inline app/code/Magento/Search/Test/Mftf/ActionGroup/AssertStorefrontVerifySearchButtonIsNotDisabledActionGroup.xml]:
```suggestion
    <actionGroup name="AssertStorefrontVerifySearchButtonIsEnabledActionGroup">
```

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @konarshankar07, thanks for the pull request, it looks great! Please follow some small suggestions below:

**Comment by konarshankar07** [comment]
@magento run all tests

**Review by gabrieldagama — APPROVED** [review]
Thanks for the updates @konarshankar07. The failed tests aren't related to this PR.

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8067](https://jira.corp.magento.com/browse/ENGCOM-8067) has been created to process this Pull Request

**Comment by konarshankar07** [comment]
@magento run Functional Tests CE test

**Comment by gabrieldagama** [comment]
@magento run Functional Tests CE

**Comment by konarshankar07** [comment]
@magento run Functional Tests CE

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8067](https://jira.corp.magento.com/browse/ENGCOM-8067) has been created to process this Pull Request

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by konarshankar07** [comment]
@magento run WebAPI Tests

**Pull request merged**